client/state/checklist/reducer.js

```diff
@@ -1,28 +1,58 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
 /**
  * Internal dependencies
  */
 import { combineReducers, keyedReducer } from 'state/utils';
 import {
 	JETPACK_MODULE_ACTIVATE_SUCCESS,
+	JETPACK_MODULE_DEACTIVATE_SUCCESS,
 	SITE_CHECKLIST_RECEIVE,
 	SITE_CHECKLIST_TASK_UPDATE,
 } from 'state/action-types';
 import { items as itemSchemas } from './schema';
 
-const markChecklistTaskComplete = ( state, taskId ) => ( {
+const toggleChecklistTask = ( state, taskId ) => ( {
 	...state,
-	tasks: { ...state.tasks, [ taskId ]: true },
+	tasks: {
+		...state.tasks,
+		[ taskId ]: { completed: ! get( state.tasks, [ taskId, 'completed' ], false ) },
+	},
 } );
 
+const moduleTaskMap = {
+	lazy_images: 'jetpack_lazy_images',
+	monitor: 'jetpack_monitor',
+	// Both photon and photon-cdn mark the Site Accelerator Task as completed
+	photon: 'jetpack_site_accelerator',
+	'photon-cdn': 'jetpack_site_accelerator',
```

Comment:
Will this setup be flexible if we decided that both features need to be enabled for the task to be marked complete?

Reply:
No, we'd have the same issue as when disabling one of the modules:
https://github.com/Automattic/wp-calypso/pull/33710#discussion_r291356465

We can work around that but will need a different solution. In this situation, we can't know the status of the unchanged module.